### PR TITLE
Fix memory reporting on FreeBSD

### PR DESCRIPTION
**Pull request by michbsd**
and move (self *Mem) function from sigar_linux_common to sigar_linux

### CODE REVIEW

**Comment by jsoriano** [comment]
@michbsd thanks for this PR, travis found some formatting errors, can you take a look? You can fix them with `gofmt -w .`.

File: sigar_freebsd.go [hunk 1]
```diff
@@ -111,3 +111,48 @@ func parseCpuStat(self *Cpu, line string) error {
...
+	}
+	pagesize := uint64(val)
+
+	name = C.CString("vm.stats.vm.v_free_count")
+	_, err = C.sysctlbyname(name, unsafe.Pointer(&val), &sc, nil, 0)
+	C.free(unsafe.Pointer(name))
+	if err != nil {
+		return err
+	}
+	self.Free = uint64(val)
```
Comment (jsoriano) [inline sigar_freebsd.go]: Is this value the number of pages? `Free` is expected to be in bytes, we should probably multiply here by pagesize.

Comment (michbsd) [reply sigar_freebsd.go]: it's done later down:

self.Used = self.Total - (self.Free * pagesize)

Comment (jsoriano) [reply sigar_freebsd.go]: But I mean to store the size in bytes in self.Free, i.e:
```
self.Free = uint64(val) * pagesize
```

`self.Free` is the value that will be used at the end for the free memory metrics.

Comment (michbsd) [reply sigar_freebsd.go]: gotcha. fixed

Comment (jsoriano) [reply sigar_freebsd.go]: Thanks, it looks good now :)

**Comment by michbsd** [comment]
ping - any news on this? 
thanks

**Comment by jsoriano** [comment]
@michbsd merged, sorry for the delay!

**Pull request merged**